### PR TITLE
Update Rust API generation (async / .await)

### PR DESCRIPTION
**Pull request by cak**
Update Rust API:
- to utilize async / .await
   - request v.10
- change parameters to string slices

Rust API PR: https://github.com/zaproxy/zap-api-rust/pull/8

### CODE REVIEW

**Comment by thc202** [comment]
Rebased to target `main` branch.

File: zap/src/main/java/org/zaproxy/zap/extension/api/RustAPIGenerator.java [hunk 1]
```diff
@@ -93,8 +93,7 @@ private void generateRustElement(ApiElement element, String component, String ty
             out.write("/**\n");
             out.write(" * " + desc + "\n");
             if (isOptional()) {
-                out.write(" * <p>\n");
-                out.write(" * " + OPTIONAL_MESSAGE + "\n");
+                out.write(OPTIONAL_MESSAGE + "\n");
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/extension/api/RustAPIGenerator.java:96]: Was this changed on purpose? It generates the following now:
```rust
/**
 * Endpoint description.
This component is optional and therefore the API will only work if it is installed
*/
```
I'd expect the ` * ` before the optional message.

**Comment by thc202** [comment]
The code changes look ok to me, though I don't know rust :)

File: zap/src/main/java/org/zaproxy/zap/extension/api/RustAPIGenerator.java (hunk 1 above)
Comment (thc202) [reply zap/src/main/java/org/zaproxy/zap/extension/api/RustAPIGenerator.java:96]: Also, the other newline would help separate the optional message from the description.

Comment (Leyart) [reply zap/src/main/java/org/zaproxy/zap/extension/api/RustAPIGenerator.java:96]: I agree it is missing the " * " in front of the optional message

**Review by Leyart — CHANGES_REQUESTED** [review]
Add " * " again in front of OPTIONAL_MESSAGE

**Pull request closed without merging**